### PR TITLE
Cobra CLI, Client Library (jsonRpc2), GSTProducer 

### PR DESCRIPTION
**Pull request by billylindeman**
- Migrates to cobra cli tool instead of using flags
- Implements an ion-sdk client for the websocket+jsonrpc2 interface
- I also ported over the gstreamer producer example from https://github.com/pion/rtwatch/ (used to exist partially in ion-load-tool)
- Fixed some issues with sample rates & added encoder customization

### CODE REVIEW

File: cfgs/local.toml [hunk 1]
```diff
@@ -14,7 +14,7 @@ key = ""
 cert = ""
 
 [signal.auth]
-enabled = true
+enabled = false
```
Comment (timsu) [inline cfgs/local.toml:17]: false by default?

**Review by timsu — APPROVED** [review]
nice! seems like mostly testing utility code

File: cfgs/local.toml (hunk 1 above)
Comment (billylindeman) [reply cfgs/local.toml:17]: this is the local dev config file so I think its fine :) helm generates the toml file from a template with params when running in k8s

File: cmd/root.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+		log.Errorf("config file %s loaded failed. range port must be [min,max]\n", cfgFile)
+		os.Exit(1)
+	}
+
+	// if len(conf.SFU.WebRTC.ICEPortRange) != 0 && conf.SFU.WebRTC.ICEPortRange[1]-conf.SFU.WebRTC.ICEPortRange[0] < portRangeLimit {
+	// 	log.Errorf("config file %s loaded failed. range port must be [min, max] and max - min >= %d\n", file, portRangeLimit)
+	// 	os.Exit(1)
+	// }
+
+	// if host := os.Getenv("ION_CLUSTER_HOST"); host != "" {
```
Comment (billylindeman) [inline cmd/root.go]: This needs to be re-enabled or ideally switched to a viper env override

**Pull request merged**